### PR TITLE
restore-util: add on split hook

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Add a hook that will be called before splitting a range.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

### CODE REVIEW

File: pkg/restore-util/split.go [hunk 1]
```diff
@@ -57,6 +66,8 @@ func (rs *RegionSplitter) Split(ctx context.Context, ranges []Range, rewriteRule
 		if rg == nil {
 			return false
 		}
+		onSplit(rg)
```
Comment (kennytm) [inline pkg/restore-util/split.go]:
```suggestion
		if onSplit != nil {
			onSplit(rg)
		}
```

**Review by kennytm — COMMENTED** [review]
Rest LGTM

File: pkg/restore-util/split.go (hunk 1 above)
Comment (overvenus) [reply pkg/restore-util/split.go]: Thanks!

**Review by 5kbpers — APPROVED** [review]
LGTM

**Pull request merged**